### PR TITLE
[SSP] Accessing __stack_chk_guard when using LTO

### PR DESCRIPTION
**Pull request by wolfy1961**
This is a pre-commit test of accessing the variable __stack_chk_guard when the static relocation model is imposed on a module compiled with pic enabled. It confirms issue [#64999](https://github.com/llvm/llvm-project/issues/64999).

The intent is to update this test with the fix for the aforementioned issue.

### CODE REVIEW

File: llvm/test/LTO/ARM/ssp-static-reloc.ll [hunk 1]
```diff
@@ -0,0 +1,40 @@
+; Confirm that we do generate one too many indirections accessing the stack guard
+; variable, when the relocation model is static.
+; This is preparation for the fix. 
+;
+; RUN: llvm-as < %s > %t.bc
+; RUN: llvm-lto -O0 -relocation-model=static -o %t.o %t.bc
+; RUN: llvm-objdump -d -r %t.o | FileCheck %s
+
```
Comment (RKSimon) [inline llvm/test/LTO/ARM/ssp-static-reloc.ll:8]: (style) Put the RUN lines at the top and the description below them

File: llvm/test/LTO/ARM/ssp-static-reloc.ll [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+entry:
+  %call = call arm_aapcscc i8 @foo()
+  %conv = zext i8 %call to i32
+  ret i32 %conv
+}
+
+attributes #0 = { noinline nounwind optnone sspstrong }
+
+!llvm.module.flags = !{!0}
+!0 = !{i32 8, !"PIC Level", i32 2}
```
Comment (RKSimon) [inline llvm/test/LTO/ARM/ssp-static-reloc.ll:40]: is all this meta data necessary?

Comment (wolfy1961) [reply llvm/test/LTO/ARM/ssp-static-reloc.ll:40]: Done

Comment (wolfy1961) [reply llvm/test/LTO/ARM/ssp-static-reloc.ll:40]: We need sspstrong and PIC Level to demonstrate the problem, i.e. the guard variable is accessed like we're generating PIC code even though there is no GOT.

**Review by RKSimon — APPROVED** [review]
LGTM

**Pull request merged**